services/src/tpu_utils/tpu_service.rs

```diff
@@ -127,7 +128,7 @@ impl TpuService {
             .get_slot_leaders(current_slot, last_slot)
             .await?;
         // get next leader with its tpu port
-        let connections_to_keep = next_leaders
+        let connections_to_keep: HashMap<_, _> = next_leaders
```

Comment:
why do you need that?
maybe you could specify the full type in that case

Reply:
There is no method to collect as Vec. Specifying HashMap should be fine here.